tests/unit_tests/blas/level1/rotmg.cpp

```diff
@@ -137,8 +137,52 @@ int test(device* dev, oneapi::mkl::layout layout) {
     auto x1_accessor = x1_buffer.template get_host_access(read_only);
     bool good_x1 = check_equal(x1_accessor[0], x1_ref, error_mag, std::cout);
     auto param_accessor = param_buffer.template get_host_access(read_only);
-    bool good_param = check_equal_vector(param_accessor, param_ref, 5, 1, error_mag, std::cout);
-    bool good = good_d1 && good_d2 && good_x1 && good_param;
+
+    constexpr fp unit_matrix = -2;
+    constexpr fp rescaled_matrix = -1;
+    constexpr fp sltc_matrix = 0;
+    constexpr fp clts_matrix = 1;
```

Comment:
Could you explain why you names these LTC and LTS? `param` is not lower triangular, right?

Reply:
The sltc / clts terminology comes from this paper:  "Hopkins, Tim (1998) Restructuring the BLAS Level 1 Routine for Computing the Modified Givens Transformation. Technical report. , Canterbury, Kent, UK."

I believe the meaning is "sin lower than cosin" and "cosin lower than sin". They are a reference to the conditions in the algorithm that define this matrixes. This is the relevant part of the algorithm that can be found at the end of the paper (in Fortran) :

`
! |c| > |s|; type zero matrix (diagonal elements one)
IF (abs(sq1)>abs(sq2)) THEN
	sflag = sltc
	sh11 = one
	sh22 = one
	sh21 = -sy1/sx1
	sh12 = sp2/sp1
	su = one - sh12*sh21
	! If su has underflowed -- sparam has already been set -- exit
	IF (su<=zero) THEN
		sd1 = zero
		sd2 = zero
		sx1 = zero
		sflag = error
	ELSE
		sd1 = sd1/su
		sd2 = sd2/su
		sx1 = sx1*su
	END IF
ELSE
! |s| >= |c|; type 1 matrix (antidiagonal case)
	IF (sq2<zero) THEN
		sd1 = zero
		sd2 = zero
		sx1 = zero
		sflag = error
	ELSE
		sflag = clts
		sh21 = -one
		sh12 = one
		sh11 = sp1/sp2
		sh22 = sx1/sy1
`

Reply:
@fabiomestre Thanks a lot for the explanation and the reference paper. It is educative :)